core-lib/src/main/java/org/idpass/smartscanner/lib/ocr/OCRAnalyzer.kt

```diff
@@ -0,0 +1,257 @@
+package org.idpass.smartscanner.lib.ocr
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Intent
+import android.graphics.Bitmap
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import android.view.View
+import android.widget.ImageView
+import androidx.camera.core.ImageProxy
+import com.google.gson.Gson
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.text.TextRecognition
+import com.google.mlkit.vision.text.latin.TextRecognizerOptions
+import org.idpass.smartscanner.api.ScannerConstants
+import org.idpass.smartscanner.lib.R
+import org.idpass.smartscanner.lib.SmartScannerActivity
+import org.idpass.smartscanner.lib.scanner.BaseImageAnalyzer
+import org.idpass.smartscanner.lib.scanner.config.ImageResultType
+import org.idpass.smartscanner.lib.scanner.config.Modes
+import org.idpass.smartscanner.lib.utils.BitmapUtils
+import org.idpass.smartscanner.lib.utils.extension.cacheImagePath
+import org.idpass.smartscanner.lib.utils.extension.cacheImageToLocal
+import org.idpass.smartscanner.lib.utils.extension.cropCenter
+import org.idpass.smartscanner.lib.utils.extension.encodeBase64
+import org.idpass.smartscanner.lib.utils.extension.isImageBlur
+import org.idpass.smartscanner.lib.utils.extension.setBrightness
+import org.idpass.smartscanner.lib.utils.extension.setContrast
+import java.io.File
+
+open class OCRAnalyzer(
```

Comment:
Is this generic OCR? or is it for a specific type of document?
Is regex optional?
What is the country for?

Reply:
OCR is generic.
Regex is optional and by default OCR will scan all texts.
Country was initially added for scope. I will try to check if this is being used. Otherwise this can be removed.

Reply:
Country is removed.